go/arrow/util/protobuf_reflect.go

```diff
@@ -401,24 +396,40 @@ func (pmr protobufMapReflection) generateKeyValuePairs() chan protobufMapKeyValu
 
 	go func() {
 		defer close(out)
-		for _, k := range pmr.rValue.MapKeys() {
+		if !pmr.rValue.IsValid() {
 			kvp := protobufMapKeyValuePairReflection{
 				k: ProtobufFieldReflection{
 					parent:        pmr.parent,
 					descriptor:    pmr.descriptor.MapKey(),
-					prValue:       getMapKey(k),
-					rValue:        k,
 					schemaOptions: pmr.schemaOptions,
 				},
 				v: ProtobufFieldReflection{
 					parent:        pmr.parent,
 					descriptor:    pmr.descriptor.MapValue(),
-					prValue:       pmr.prValue.Map().Get(protoreflect.MapKey(getMapKey(k))),
-					rValue:        pmr.rValue.MapIndex(k),
 					schemaOptions: pmr.schemaOptions,
 				},
 			}
 			out <- kvp
+		} else {
+			for _, k := range pmr.rValue.MapKeys() {
```

Comment:
May be a bit nicer instead to `return` early from the conditional and get rid of the need for `else`, since this block already has considerable nesting.

```suggestion
			return
		}
		for _, k := range pmr.rValue.MapKeys() {
```